client/login/magic-login/index.jsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CHECK_YOUR_EMAIL_PAGE,
+	INTERSTITIAL_PAGE,
+	LINK_EXPIRED_PAGE,
+} from 'state/login/magic-login/constants';
+import { isEnabled } from 'config';
+import EmailedLoginLinkSuccessfully from '../magic-login/emailed-login-link-successfully';
+import EmailedLoginLinkExpired from '../magic-login/emailed-login-link-expired';
+import {
+	getMagicLoginEmailAddressFormInput,
+	getMagicLoginCurrentView,
+} from 'state/selectors';
+import { getCurrentQueryArguments } from 'state/ui/selectors';
+import HandleEmailedLinkForm from '../magic-login/handle-emailed-link-form';
+import {
+	hideMagicLoginRequestForm,
+	showMagicLoginInterstitialPage,
+} from 'state/login/magic-login/actions';
+import Main from 'components/main';
+import RequestLoginEmailForm from '../magic-login/request-login-email-form';
+import { recordTracksEvent } from 'state/analytics/actions';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import GlobalNotices from 'components/global-notices';
+import notices from 'notices';
+import { login } from 'lib/paths';
+
+class MagicLogin extends React.Component {
+	static propTypes = {
+		hideMagicLoginRequestForm: PropTypes.func.isRequired,
+		magicLoginEmailAddress: PropTypes.string,
+		magicLoginEnabled: PropTypes.bool,
+		magicLoginView: PropTypes.string,
+		recordTracksEvent: PropTypes.func.isRequired,
+		showMagicLoginInterstitialPage: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	onClickEnterPasswordInstead = event => {
+		event.preventDefault();
+		this.props.recordTracksEvent( 'calypso_login_enter_password_instead_click' );
+
+		page( login( { isNative: true } ) );
+	};
+
+	magicLoginMainContent() {
+		const {
+			magicLoginView,
+			magicLoginEmailAddress,
+		} = this.props;
+
+		switch ( magicLoginView ) {
+			case LINK_EXPIRED_PAGE:
+				this.props.recordTracksEvent( 'calypso_login_magic_link_expired_link_view' );
```

Comment:
Again, I understand the main point of the PR is refactoring, so consider this for later:

Semantically, these are events tracking a page view, so the `PageView` analytics component could be more appropriate. Edit: ah, I see it's actually being used in the `render`